src/models/Applicant/Section/Repository.ts

```diff
@@ -2,13 +2,35 @@ import { Applicant } from "../Model";
 import { Section } from ".";
 import { omit } from "lodash";
 import { TSection } from "../Interface";
+import { Op } from "sequelize";
+import isEmpty from "lodash/isEmpty";
+import { Transaction } from "sequelize";
 
 export const SectionRepository = {
-  updateOrCreate: async (applicant: Applicant, section: TSection) => {
+  update: async (applicant: Applicant, sections: TSection[], transaction?: Transaction) => {
+    const sectionsUuid: string[] = [];
+    for (const section of sections) {
+      sectionsUuid.push(await SectionRepository.updateOrCreate(applicant, section, transaction));
+    }
+    return Section.destroy({
+      where: {
+        applicantUuid: applicant.uuid,
+        ...(!isEmpty(sectionsUuid) && {
+          [Op.not]: {
+            uuid: sectionsUuid
+          }
+        })
+      },
+      transaction
+    });
+  },
+  updateOrCreate: async (applicant: Applicant, section: TSection, transaction?: Transaction) => {
```

Comment:
si lo que updateo es la section, lo pondría como primer parámetro
así se leería mas como
"update a section, with an applicant and a transaction"